src/utils/web-preset.ts

```diff
@@ -9,52 +10,66 @@ import { isUndefined } from './helper'
  */
 let online = true
 const isOnline = () => online
+const createNoop = (ret?: any) => () => ret
+const add = 'addEventListener'
+const remove = 'removeEventListener'
 
-// For node and React Native, `window.addEventListener` doesn't exist.
-const addWindowEventListener =
-  typeof window !== 'undefined' && !isUndefined(window.addEventListener)
-    ? window.addEventListener.bind(window)
-    : null
-const addDocumentEventListener =
-  typeof document !== 'undefined'
-    ? document.addEventListener.bind(document)
-    : null
+type EventAction = 'addEventListener' | 'removeEventListener'
+
+const hasWindow = typeof window !== 'undefined'
+const hasDocument = typeof document !== 'undefined'
+
+// For node and React Native, `add/removeEventListener` doesn't exist on window.
+const windowEventListener = (action: EventAction) =>
+  hasWindow && !isUndefined(window[action])
+    ? window[action].bind(window)
+    : createNoop()
+const documentEventListener = (action: EventAction) =>
+  hasDocument ? document[action].bind(document) : createNoop()
 
 const isDocumentVisible = () => {
-  if (addDocumentEventListener) {
-    const visibilityState = document.visibilityState
-    if (!isUndefined(visibilityState)) {
-      return visibilityState !== 'hidden'
-    }
+  const visibilityState = hasDocument && document.visibilityState
+  if (!isUndefined(visibilityState)) {
+    return visibilityState !== 'hidden'
   }
-  // always assume it's visible
   return true
 }
 
-const registerOnFocus = (cb: () => void) => {
-  if (addWindowEventListener && addDocumentEventListener) {
-    // focus revalidate
-    addDocumentEventListener('visibilitychange', cb)
-    addWindowEventListener('focus', cb)
+const setupOnFocus = (cb: () => void) => {
+  // focus revalidate
+  documentEventListener(add)('visibilitychange', cb)
+  windowEventListener(add)('focus', cb)
+  return () => {
+    documentEventListener(remove)('visibilitychange', cb)
+    windowEventListener(remove)('focus', cb)
   }
```

Comment:
If we are setting up those listeners globally instead of doing it with a hook, there's no need to return unregisters I think.